src/providers/ldap/sdap_async.c

```diff
@@ -616,7 +678,13 @@ struct tevent_req *sdap_exop_modify_passwd_send(TALLOC_CTX *memctx,
     DEBUG(SSSDBG_TRACE_INTERNAL,
           "ldap_extended_operation sent, msgid = %d\n", msgid);
 
-    ret = sdap_op_add(state, ev, state->sh, msgid,
+    stat_info = talloc_asprintf(state, "server: modify passwd dn: [%s]",
+                                sdap_get_server_ip_str(state->sh), user_dn);
```

Comment:
```
Error: COMPILER_WARNING (CWE-685):
sssd-2.6.3/src/providers/ldap/sdap_async.c: scope_hint: In function 'sdap_exop_modify_passwd_send'
sssd-2.6.3/src/providers/ldap/sdap_async.c:681:40: warning[-Wformat-extra-args]: too many arguments for format
#  681 |     stat_info = talloc_asprintf(state, "server: modify passwd dn: [%s]",
#      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#  679|             "ldap_extended_operation sent, msgid = %d\n", msgid);
```

Reply:
Btw, can `talloc_asprintf()` handle NULL for `%s` gracefully?

Reply:
Fixed and `talloc_asprintf()` can handle `NULL` gracefully as long as `vsnprintf()` of the underlying C library can handle it gracefully.

Reply:
> Fixed and `talloc_asprintf()` can handle `NULL` gracefully as long as `vsnprintf()` of the underlying C library can handle it gracefully.

But it (`vsnprintf()`) doesn't have to (and we check for NULL explicitly in a lot of places through the code).

Reply:
But I guess there is a similar amount of places where we rely on glibc/vsnprinft() to handle it gracefully.

Reply:
This is UB. And this PR has this check in `sdap_call_op_callback()`.

Reply:
(JFTR: I mean return value of `sdap_get_server_ip_str()` here)

Reply:
Hi,

I added a new call which never returns `NULL` in the latest version.

bye,
Sumit

Reply:
Thank you.